topeft/modules/datacard_tools.py

```diff
@@ -847,22 +847,43 @@ def analyze(self,km_dist,ch,selected_wcs, crop_negative_bins):
                         "rate": -1
                     }
                     # There should be only 1 sparse axis at this point, the systematics axis
+                    check_zero_arr0 = False
+                    check_zero_arr1 = False
                     for sp_key,arr in v.items():
+                        syst = sp_key[0]
                         if crop_negative_bins:
                             negative_bin_mask = np.where( arr[0] < 0) # see where bins are negative
                             arr[0][negative_bin_mask] = np.zeros_like( arr[0][negative_bin_mask] )  # set those to zero
                             if arr[1] is not None:
                                 arr[1][negative_bin_mask] = np.zeros_like( arr[1][negative_bin_mask] )  # if there's a sumw2 defined, that one's set to zero as well. Otherwise we will get 0 +/- something, which is compatible with negative
 
-                        syst = sp_key[0]
+                        if len(arr[0][negative_bin_mask]) and syst =="nominal":  # check systematics error for fake factors
+                            if sum(arr[0]) == 0:
+                                check_zero_arr0 = True
+                            if sum(arr[1]) == 0:
+                                check_zero_arr1 = True
+                        if "Up" in syst:
```

Comment:
@ywan2 after thinking about the terms a bit more, I still think it's important to check _all_ systematics, but with two caveats: 
- The `FF` terms _only_ affect the `fakes_sm` process
- The other terms (e.g. `FSR`, `renorm`, ...) affect every process _but_ `fakes_sm`

Something like this might be the minimal changes needed:
```suggestion
                        check_zero_arr0 = check_zero_arr0 and ('fakes' in proc and 'FF' in syst) or ('fakes' not in proc and 'FF' not in syst)
                        check_zero_arr1 = check_zero_arr1 and ('fakes' in proc and 'FF' in syst) or ('fakes' not in proc and 'FF' not in syst)
                        if "Up" in syst:
```
Now I don't think `proc` is defined here, so see if there's an easy way to pass it or get it here.